articlewebview.cc

```diff
@@ -60,84 +57,125 @@ void ArticleWebView::triggerPageAction( QWebPage::WebAction action, bool checked
   }
 #endif
 
-  QWebView::triggerPageAction( action, checked );
+  QWebEngineView::triggerPageAction( action, checked );
 }
 
 bool ArticleWebView::event( QEvent * event )
 {
-  switch ( event->type() )
-  {
-  case QEvent::MouseButtonRelease:
-  case QEvent::MouseButtonDblClick:
-    showInspectorDirectly = true;
-    break;
+    if (event->type() == QEvent::ChildAdded) {
+        QChildEvent *child_ev = static_cast<QChildEvent *>(event);
 
-  case QEvent::ContextMenu:
-    showInspectorDirectly = false;
-    break;
+//      // there is also QObject child that should be ignored here;
+//      // use only QOpenGLWidget child
+//      QOpenGLWidget *w = static_cast<QOpenGLWidget*>(child_ev->child());
 
-  default:
-    break;
-  }
+        child_ev->child()->installEventFilter(this);
+    }
+
+    return QWebEngineView::event(event);
+}
+
+void ArticleWebView::linkClickedInHtml(QUrl const& ){
+  //disable single click to simulate dbclick action on the new loaded pages.
+  singleClickToDbClick=false;
+}
+
+bool ArticleWebView::eventFilter(QObject *obj, QEvent *ev) {
+    if (ev->type() == QEvent::MouseButtonDblClick) {
+      QMouseEvent *pe = static_cast<QMouseEvent *>(ev);
```

Comment:
Totally wrong code formatting here and in many other places. Consider running automatic code formatting tools on your changes, e.g. clang-format. Or just go over everything and fix the formatting manually, if you prefer. You could use the occasion to review your changes and remove unrelated diff.

Reply:
The formatting in this file is still wrong (parentheses must be padded with spaces inside, see existing code). The formatting of most code in other files is wrong too.